StackExchange.Profiling/Timing.cs

```diff
@@ -299,6 +325,7 @@ internal void RemoveCustomTiming(string category, CustomTiming customTiming)
         }
 
         private readonly object _lockObject = new object();
```

Comment:
Not your code, but I don't understand the locking around CustomTimingList. 

The existing `GetCustomTimingList` acquires the same lock twice, and protects access to the dictionary. But the underlying values in the dictionary (lists of custom timings) are not protected by the lock. I don't see why they would need to protect the dictionary but the lists.